src/image-upload/image.service.ts

```diff
@@ -5,6 +5,23 @@ import { Observable } from 'rxjs/Observable';
 @Injectable()
 export class ImageService {
   constructor(private http: Http) {
+     /*
+    Here it is important to note that the Internet explorer browser must enter the catch, execute inside can perform
+     file = new File([fileBlob], fileUrl); 
+    */
+    const YaunlaideFile = (<any>window).File;
+    (<any>window).File = function (chunks1, filename1, opts1 = {}) {
```

Comment:
I'm not sure the best way is to replace the file constructor immediately and then run the test every time its called. Surely you can test the file constructor once, then replace if necessary.

```typescript
try {
  new File([], '');
} catch (error) {
  (<any>window).File = function (chunks, filename, options = {}) {
    Blob.call(this, chunks, options);
    (<any>this).lastModifiedDate = new Date();
    (<any>this).lastModified = + (<any>this).lastModifiedDate;
    (<any>this).name = filename;
  }
}
```

I think this should be wrapped in a function and moved to a different file, and it should be called elsewhere - perhaps in `ImageUploadComponent` as this is the class that actually calls the `File` constructor.